integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/TaskFhirResourceProviderIntegrationTest.java

```diff
@@ -0,0 +1,325 @@
+		
+		assertThat(response, isNotFound());
+		assertThat(response.getContentType(), is(BaseFhirIntegrationTest.FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+	}
+	
+	@Test
+	public void shouldCreateNewTaskAsJson() throws Exception {
+		String jsonTask;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_CREATE_TASK_PATH)) {
+			Objects.requireNonNull(is);
+			jsonTask = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		// create task
+		MockHttpServletResponse response = post("/Task").accept(FhirMediaTypes.JSON).jsonContent(jsonTask).go();
+		//lets verify the created task 
+		assertThat(response, isCreated());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+	Task task = readResponse(response);
+		assertThat(task, notNullValue());
+		
+		// try to get new task
+		response = get(task.getId()).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		
+		Task task2 = readResponse(response);
+		
+		assertThat(task2.getId(), equalTo(task.getId()));
+		
+	}
+	
+	@Test
+	public void shouldCreateNewTaskAsXml() throws Exception {
+		
+		String xmlTask;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(XML_CREATE_TASK)) {
+			Objects.requireNonNull(is);
+			xmlTask = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
+		// create task
+		MockHttpServletResponse response = post("/Task").accept(FhirMediaTypes.XML).xmlContext(xmlTask).go();
+		// verify created correctly
+		
+		assertThat(response.getContentType(), is(FhirMediaTypes.XML.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Task task = readResponse(response);
+		assertThat(task, notNullValue());
+		assertThat(task.getStatus(), is(Task.TaskStatus.ACCEPTED));
+	
+		
+		// try to get new task
+		response = get(task.getId()).accept(FhirMediaTypes.XML).go();
+		
+		assertThat(response, isOk());
+		
+		Task newTask = readResponse(response);
+		
+		assertThat(newTask.getId(), equalTo(task.getId()));
+		assertThat(newTask.getStatus(), equalTo(task.getStatus()));
+	}
+	
+	@Test
+	public void shouldUpdateExistingTaskAsJson() throws Exception {
+		//Before update
+		MockHttpServletResponse response = get("/Task/" + TASK_UUID).accept(FhirMediaTypes.JSON).go();
+		
+		assertThat(response, isOk());
+		assertThat(response.getContentType(), is(FhirMediaTypes.JSON.toString()));
+		assertThat(response.getContentAsString(), notNullValue());
+		
+		Task task = readResponse(response);
+//		
+//		assertThat(task, notNullValue());
+//		assertThat(task, validResource());
+//		assertThat(task.getIdElement().getIdPart(), equalTo(TASK_UUID));
+		
+		String jsonTask;
+		try (InputStream is = this.getClass().getClassLoader().getResourceAsStream(JSON_UPDATE_TASK_PATH)) {
+			Objects.requireNonNull(is);
+			jsonTask = IOUtils.toString(is, StandardCharsets.UTF_8);
+		}
```

Comment:
We shouldn't need to read a JSON file here. See [how this is implemented in the tests for Patient](https://github.com/openmrs/openmrs-module-fhir2/blob/20d9b015c6bb6d9a428ad3947272ba56c1b6498a/integration-tests/src/test/java/org/openmrs/module/fhir2/providers/r3/PatientFhirResourceProviderIntegrationTest.java#L222).

Reply:
Thanks @ibacher